Major redesign for 1.0.0.pre

bliof opened the pull request:
Changes in the logic for selecting thumbs:
- when trying to select a retina thumb it will try to select one that
  is >= to the wanted size and if there isn't such will fallback to
  normal thumbs
- things like '320x' will select sizes like '420x500' only if the
  isn't anything like '640x' aka larger than '320x'

Adds more places in which you could change the logic of how to select a
thumb, e.g. you could overwrite artwork_thumbs_for so that it return
thumbs with width and height that you have save for example in a db.

Rename artwork_thumb_for -> attachment_style_for
NB. there is still an artwork_thumb_for method that selects only a compatible artwork thumb/style
The attachment_style_for is combination of artwork_thumb_for and plain_thumb_for.
The second handles noncompatible styles.

bliof committed:
Major redesign for 1.0.0.pre

Changes in the logic for selecting thumbs:
  - when trying to select a retina thumb it will try to select one that
    is >= to the wanted size and if there isn't such will fallback to
    normal thumbs
  - things like '320x' will select sizes like '420x500' only if the
    isn't anything like '640x' aka larger than '320x'

Adds more places in which you could change the logic of how to select a
thumb, e.g. you could overwrite artwork_thumbs_for so that it return
thumbs with width and height that you have save for example in a db.

Rename artwork_thumb_for -> attachment_style_for
NB. there is still an artwork_thumb_for method that selects only an compatible artwork thumb/style
The attachment_style_for is combination of artwork_thumb_for and plain_thumb_for.
The second handles noncompatible styles.

mitio commented on lib/artwork/model.rb:
```diff
@@ -1,60 +1,90 @@
 
-      size, base_resolution = size.to_s.split('@')
-      base_resolution ||= Artwork.base_resolution
+      result =
```
`result` -> `thumb`

mitio commented on lib/artwork/model.rb:
```diff
@@ -1,60 +1,90 @@
-    def artwork_url(attachment_name, size, options = {})
-      thumb_name = artwork_thumb_for(attachment_name, size, options)
-      send(attachment_name).url(thumb_name, options)
+    def artwork_normal_thumb_for(attachment_name, thumbs, desired_thumb)
```
`artwork_normal_thumb_for` -> `normal_artwork_thumb_for`?

bliof committed:
Return the exact style when you search for '320x_2x'

Change DesiredThumbnail so that is does not understand retina
prefixes aka they are incompatile. By doing this when you do something
like user.artwork_url(:avatar, '320x_2x') you will always get the url
for the 320x_2x style.
Move the logic that scales the width into the current resolution in
DesiredThumbnail, because it contains the base_resolution.
Refactor a bit the DesiredThumbnail.
Add a bit of specs.
Add Artwork::expected_width_for that is a proxy to DesiredThumbnail#width_in_current_resolution

bliof committed:
Pick better name for a local variable in attachment_style_for

bliof committed:
Keep the 'artwork_thumb_for' stem in the methods that pick retina and normal thumbs

artwork_normal_thumb_for -> normal_artwork_thumb_for
artwork_retina_thumb_for -> retina_artwork_thumb_for

bliof committed:
Return the largest retina_image if nothing was found in retina_artwork_thumb_for

bliof committed:
Refactor a bit the Artwork::Model

mitio committed:
Update the changelog for v1.0.0

mitio committed:
v1.0.0
